crates/cairo-addons/src/vm/dict_manager.rs

```diff
@@ -47,24 +123,22 @@ impl PyDictManager {
 }
 
 #[pyclass(name = "DictTracker")]
+#[derive(Clone)]
 pub struct PyDictTracker {
     inner: DictTracker,
 }
 
 #[pymethods]
 impl PyDictTracker {
-    // Note: This is a temporary implementation, need to understand why HashMap<PyMaybeRelocatable,
-    // PyMaybeRelocatable> is not working
     #[new]
-    #[pyo3(signature = (keys, values, current_ptr, default_value=None))]
+    #[pyo3(signature = (data, current_ptr, default_value=None))]
     fn new(
```

Comment:
this look like a from method rather than a new?

Reply:
what?

Reply:
idiomatically, it looks like this function could be called `from_py_dict` or smth that suggests the hashmap data exists, with keys and values of python types

Reply:
no, it's a pyo3 constructor